Deprecated class warning

Using GCC 12 on Fedora:
```cpp
[12/361] Building CXX object Macaulay2...hic/CMakeFiles/mathic.dir/mathic.cpp.o
In file included from /home/mahrud/Projects/M2/engine/M2/Macaulay2/e/mathic/mathic.h:13,
                 from /home/mahrud/Projects/M2/engine/M2/Macaulay2/e/mathic/mathic.cpp:1:
/home/mahrud/Projects/M2/engine/M2/Macaulay2/e/mathic/mathic/DivList.h:163:15: warning: ‘template<class _Category, class _Tp, class _Distance, class _Pointer, class _Reference> struct std::iterator’ is deprecated [-Wdeprecated-declarations]
  163 |   public std::iterator<std::bidirectional_iterator_tag, Entry> {
      |               ^~~~~~~~
In file included from /usr/include/c++/12/string:45,
                 from /usr/include/c++/12/bits/locale_classes.h:40,
                 from /usr/include/c++/12/bits/ios_base.h:41,
                 from /usr/include/c++/12/ios:42,
                 from /usr/include/c++/12/ostream:38,
                 from /home/mahrud/Projects/M2/engine/M2/Macaulay2/e/mathic/mathic/Timer.h:7,
                 from /home/mahrud/Projects/M2/engine/M2/Macaulay2/e/mathic/mathic.h:4:
/usr/include/c++/12/bits/stl_iterator_base_types.h:127:34: note: declared here
  127 |     struct _GLIBCXX17_DEPRECATED iterator
      |                                  ^~~~~~~~
/home/mahrud/Projects/M2/engine/M2/Macaulay2/e/mathic/mathic/DivList.h:191:15: warning: ‘template<class _Category, class _Tp, class _Distance, class _Pointer, class _Reference> struct std::iterator’ is deprecated [-Wdeprecated-declarations]
  191 |   public std::iterator<std::bidirectional_iterator_tag, const Entry> {
      |               ^~~~~~~~
/usr/include/c++/12/bits/stl_iterator_base_types.h:127:34: note: declared here
  127 |     struct _GLIBCXX17_DEPRECATED iterator
      |                                  ^~~~~~~~
```

--- a/src/mathic/DivList.h
+++ b/src/mathic/DivList.h
@@ -159,9 +159,14 @@ namespace mathic {
   };
 
   template<class C>
-    class DivList<C>::iterator :
-  public std::iterator<std::bidirectional_iterator_tag, Entry> {
+    class DivList<C>::iterator {
   public:
+    typedef std::bidirectional_iterator_tag iterator_category;
+    typedef Entry value_type;
+    typedef std::ptrdiff_t difference_type;
+    typedef Entry *pointer;
+    typedef Entry &reference;
+
     friend class DivList<C>;
     explicit iterator(ListIter it): _it(it) {}
     operator const_iterator() const {return const_iterator(_it);}
@@ -187,9 +192,14 @@ namespace mathic {
   };
 
   template<class C>
-    class DivList<C>::const_iterator :
-  public std::iterator<std::bidirectional_iterator_tag, const Entry> {
+    class DivList<C>::const_iterator {
   public:
+    typedef std::bidirectional_iterator_tag iterator_category;
+    typedef const Entry value_type;
+    typedef std::ptrdiff_t difference_type;
+    typedef const Entry *pointer;
+    typedef const Entry &reference;
+
     friend class DivList<C>;
     explicit const_iterator(CListIter it): _it(it) {}
     bool operator==(const_iterator it) {return _it == it._it;}

